plugins/by-name/treesitter/default.nix

```diff
@@ -189,135 +147,42 @@ lib.nixvim.plugins.mkNeovimPlugin {
-        a lua function that returns a boolean indicating the parser should be disabled.
-      '';
-
-      custom_captures = lib.nixvim.defaultNullOpts.mkAttrsOf types.str { } ''
-        Custom capture group highlighting.
-      '';
-    };
-
-    incremental_selection = {
-      enable = lib.nixvim.defaultNullOpts.mkBool false ''
-        Incremental selection based on the named nodes from the grammar.
-      '';
-
-      keymaps =
-        let
-          mkKeymap =
-            default:
-            lib.nixvim.defaultNullOpts.mkNullableWithRaw (
-              with types; either str bool
-            ) default "Key shortcut or false to unset.";
-        in
-        {
-          init_selection = mkKeymap "gnn";
-          node_incremental = mkKeymap "grn";
-          scope_incremental = mkKeymap "grc";
-          node_decremental = mkKeymap "grm";
-        };
-    };
-
-    indent = {
-      enable = lib.nixvim.defaultNullOpts.mkBool false ''
-        Whether to enable treesitter indentation.
-      '';
-    };
-
-    ensure_installed = lib.nixvim.defaultNullOpts.mkNullable' {
-      type =
-        with lib.types;
-        oneOf [
-          (enum [ "all" ])
-          (listOf (maybeRaw str))
-          rawLua
-        ];
-      pluginDefault = [ ];
-      description = ''
-        Either `"all"` or a list of languages to ensure installing.
-      '';
-    };
-
-    ignore_install = lib.nixvim.defaultNullOpts.mkListOf types.str [ ] ''
-      List of parsers to ignore installing. Used when `ensure_installed` is set to `"all"`.
-    '';
-
-    parser_install_dir = lib.nixvim.mkNullOrOption' {
-      type = with lib.types; maybeRaw str;
-      # Backport the default from nvim-treesitter 1.0
-      # The current default doesn't work on nix, as it is readonly
-      default.__raw = "vim.fs.joinpath(vim.fn.stdpath('data'), 'site')";
-      pluginDefault = lib.literalMD "the plugin's package directory";
-      description = ''
-        Location of the parsers to be installed by the plugin (only needed when `nixGrammars` is disabled).
-
-        By default, parsers are installed to the "site" dir.
-        If set to `null` the _plugin default_ is used, which will not work on nix.
-      '';
-    };
-
-    sync_install = lib.nixvim.defaultNullOpts.mkBool false ''
-      Install parsers synchronously (only applied to `ensure_installed`).
-    '';
-  };
-
   settingsExample = {
-    auto_install = false;
-    ensure_installed = "all";
-    ignore_install = [ "rust" ];
-    parser_install_dir.__raw = "vim.fs.joinpath(vim.fn.stdpath('data'), 'treesitter')";
-    sync_install = false;
-
-    highlight = {
-      enable = true;
+    install_dir = lib.nixvim.nestedLiteralLua "vim.fs.joinpath(vim.fn.stdpath('data'), 'site')";
+  };
 
-      additional_vim_regex_highlighting = true;
-      disable = [ "rust" ];
-      custom_captures = { };
+  extraOptions = {
+    folding = mkOption {
+      type = types.either types.bool (
```

Comment:
Can we use `transitionType` here?

That'd a) handle the deprecation for us and b) avoid documenting the `either` type.

We can also use `visible = "transparent"`, so that only the sub-options (`enable`) are documented. Although, I'm not sure nixvim's docs know how to handle `visible = "transparent"`...

Reply:
> You can keep your current solution,

Do we have access to `options` in the `apply` function or how can we handle: 

>I'd also prefer we use option-string-interpolation, so the warnings include the full path, complete with prefix.